### PR TITLE
Rename DictionaryBasedValueProvider to RouteValueProvider

### LINKED ISSUE #3629: Simplify RouteValueProvider/RouteValueProviderFactory
We should rename https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.Core/ModelBinding/DictionaryBasedValueProvider.cs to `RouteValueProvider`. Our other VPs have distinct class names.

We should also add a `protected set` `CultureInfo` property to all of our VPFs so that scenarios like [this one](https://github.com/aspnet/Routing/issues/237) are easier to address.

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #3629

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @rynowak \ @Eilon

File: src/Microsoft.AspNet.Mvc.Core/ModelBinding/JQueryFormValueProvider.cs [hunk 1]
```diff
@@ -9,7 +9,7 @@
 namespace Microsoft.AspNet.Mvc.ModelBinding
 {
     /// <summary>
-    /// An <see cref="IValueProvider"/> for form data stored in an <see cref="IDictionary{string, string[]}"/>.
+    /// An <see cref="IValueProvider"/> for JQuery formatted form data.
```
Comment (Eilon) [inline src/Microsoft.AspNet.Mvc.Core/ModelBinding/JQueryFormValueProvider.cs]: `jQuery` (that's how they case it)

Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Core/ModelBinding/JQueryFormValueProvider.cs]: https://msdn.microsoft.com/en-us/library/system.web.mvc.jqueryformvalueprovider(v=vs.118).aspx

> The JQuery Form Value provider is used to handle JQuery formatted data in request Forms.

Comment (Eilon) [inline src/Microsoft.AspNet.Mvc.Core/ModelBinding/JQueryFormValueProvider.cs]: Believe it or not, MSDN is wrong :smile:

**Comment by pranavkm** [comment]
Updated

**Comment by Eilon** [comment]
:shipit:

**Comment by rynowak** [comment]
:shipit:

**Pull request closed without merging**